File: tests/fixtures/parser-compact/projects/data-shape/data-shape-basic/typescript/src/user-summary.ts
export class UserSummary {
  constructor(
    public readonly userId: number,
    public readonly name: string,
    public readonly active: boolean,
  ) {}

  label(): string {
    if (this.active) {
      return `${this.name}#${this.userId}`;
    }
    return "inactive";
  }
}

File: tests/fixtures/parser-compact/projects/data-shape/data-shape-basic/python/src/user_summary.py
"""Data shape compact fixture for parser snapshot tests."""

from dataclasses import dataclass


@dataclass(frozen=True)
class UserSummary:
    user_id: int
    name: str
    active: bool

    def label(self) -> str:
        if self.active:
            return f"{self.name}#{self.user_id}"
        return "inactive"
